websocket_server.py

```diff
@@ -0,0 +1,269 @@
+        pick_prev = min(len(self.text), self.pick_previous_segments)
+        for seg in self.text[-pick_prev:]:
+            # discard everything before a 3 second pause
+            if seg == '':
+                text = ''
+            else:
+                text += seg
+        wrapped = "".join(text + output)
+        return wrapped
+    
+    def add_frames(self, frame_np):
+        if self.frames_np is not None and self.frames_np.shape[0] > 45*self.RATE:
+            self.frames_offset += 45.0
+            self.frames_np = self.frames_np[int(30*self.RATE):]
+        if self.frames_np is None:
+            self.frames_np = frame_np.copy()
+        else:
+            self.frames_np = np.concatenate((self.frames_np, frame_np), axis=0)
+
+    def speech_to_text(self):
+        """
+        Process audio stream in an infinite loop.
+        """
+        while True:
+            if self.exit:
+                logging.info("Exiting speech to text thread")
+                break
+            
+            if self.frames_np is None: 
+                continue
+
+            # clip audio if the current chunk exceeds 30 seconds, this basically implies that
+            # no valid segment for the last 30 seconds from whisper
+            if self.frames_np[int((self.timestamp_offset - self.frames_offset)*self.RATE):].shape[0] > 25 * self.RATE:
+                duration = self.frames_np.shape[0] / self.RATE
+                self.timestamp_offset = self.frames_offset + duration - 5
+    
+            samples_take = max(0, (self.timestamp_offset - self.frames_offset)*self.RATE)
+            input_bytes = self.frames_np[int(samples_take):].copy()
+            duration = input_bytes.shape[0] / self.RATE
+            if duration<1.0: 
+                continue
+            try:
+                input_sample = input_bytes.copy()
+                # set previous complete segment as initial prompt
+                if len(self.text) and self.text[-1] != '': 
+                    initial_prompt = self.text[-1]
+                else: 
+                    initial_prompt = None
+                
+                # whisper transcribe with prompt
+                result = self.transcriber.transcribe(input_sample, initial_prompt=initial_prompt)
+                if len(result):
+                    self.t_start = None
+                    output, last_segment = self.update_segments(result, duration)
+                    if len(self.transcript) < self.send_last_n_segments:
+                        segments = self.transcript
+                    else:
+                        segments = self.transcript[-self.send_last_n_segments:]
+                    if last_segment is not None:
+                        segments = segments + [last_segment]
+                    out_dict = {
+                        'text': output,
+                        'segments': segments
+                    }
+                    try:
+                        self.websocket.send(str(out_dict))
```

Comment:
```suggestion
                        self.websocket.send(json.dumps(out_dict))
```

to make a proper json string.